src/esbuild_plugins/islands.ts

```diff
@@ -1,19 +1,26 @@
+import { normalize } from "https://deno.land/std@0.177.0/path/mod.ts";
 import type { Plugin } from "https://deno.land/x/esbuild@v0.17.16/mod.js";
 import { compile, preprocess } from "npm:svelte/compiler";
 
 const filter = /\.svelte$/;
 const name = "mononykus/svelte-islands";
 
-export const island_wrapper = (mode: "ssr" | "dom", dir: string): Plugin => ({
+export const island_wrapper = (
+	mode: "ssr" | "dom",
+	dir: string,
+	base_path: string,
+): Plugin => ({
 	name,
 	setup(build) {
 		build.onLoad({ filter }, async ({ path }) => {
 			const filename = path.split(dir).at(-1) ?? "Undefined.svelte";
 			const source = await Deno.readTextFile(path);
 			const island = filename.match(/\/(\w+).island.svelte/);
 
-			const processed = island && mode === "ssr"
-				? (await preprocess(source, {
+			let processed = source;
```

Comment:
Unless we can benchmark this and demonstrate that it is faster, I would much prefer to keep things immutable.